perf/src/sigverify.rs

```diff
@@ -390,35 +393,80 @@ pub fn generate_offsets(
     )
 }
 
-fn dedup_packet(count: &AtomicU64, packet: &mut Packet, bloom: &AtomicBloom<&[u8]>) {
-    // If this packet was already marked as discard, drop it
-    if packet.meta.discard {
-        return;
+pub struct Deduper {
+    filter: Vec<AtomicU64>,
+    seed: RandomState,
+    age: Instant,
+    max_age: Duration,
+    saturated: AtomicBool,
+}
+
+impl Deduper {
+    pub fn new(size: u32, max_age_ms: u64) -> Self {
+        let mut filter: Vec<AtomicU64> = Vec::with_capacity(size as usize);
```

Comment:
same as the vec! macro right?
`= vec![size; Default::default());`

Reply:
|                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Clone` is not implemented for `AtomicU64`

Reply:
doesn't like it

Reply:
ah, ok